src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2205,16 +2205,23 @@ primitive_inst::primitive_inst(network & network, program_node const& node, bool
             }
         }
 
-        // TODO: Remove WA for arg_max_min node.
-        // For now it's required to handle the case when only second output of TopK primitive is used in plugin,
-        // but kernels always write both outputs to the same memory object which leads to wrong result.
-        if (user_count == 1 && mutable_data_count == 1 && !node.is_type<arg_max_min>()
-                                                       && !node.is_type<experimental_detectron_roi_feature_extractor>()) {
-            for (auto& user : node.get_users())
-                if (user->is_type<mutable_data>())
-                    _outputs[0] = user->as<mutable_data>().get_attached_memory_ptr();
+        if (node.is_reusing_memory() ) {      
+                const auto& eltw_inst = _network.get_primitive(node.reuse_id());
+                auto& eltw_mem = eltw_inst->output_memory();
+                auto new_mem = eltw_mem.get_engine()->reinterpret_buffer(eltw_mem, node.get_output_layout());
+                _outputs.push_back(new_mem);
```

Comment:
Allocation order is based on the size, which might not ensure eltw input memory is already allocated at this moment.

Reply:
Looking into the details in [get_primitive](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/graph/network.cpp#L895) where it will allocate the primitive instance if it couldn't be found. Should it be enough to address the concern?

Reply:
As I mentioned in offline, this might not be helping becuase of the opt_out inputs.